### PR TITLE
[chore] use status Type in azureeventhubreceiver

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler_test.go [hunk 1]
```diff
@@ -236,7 +236,7 @@ func TestUnmarshalLogs(t *testing.T) {
 	expectedMinimum := plog.NewLogs()
 	resourceLogs := expectedMinimum.ResourceLogs().AppendEmpty()
 	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
-	scopeLogs.Scope().SetName("otelcol/" + typeStr)
+	scopeLogs.Scope().SetName("otelcol/azureeventhubreceiver")
```
Comment (codeboten) [inline receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler_test.go:239]: Should this use `receiverScopeName`?

Comment (atoulme) [reply receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler_test.go:239]: it's a test so the idea is to hardcode things so we fail loudly if things change from under us.

Comment (codeboten) [reply receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler_test.go:239]: fair enough, the previous test was using `typeStr` but i agree that hardcoding is ok here.

Comment (atoulme) [reply receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler_test.go:239]: I'm codeowner of this component and I would like to apply to the maximum pain principle here. I need things to break or I'll happily gloss over them.

File: receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler_test.go [hunk 2]
```diff
@@ -236,7 +236,7 @@ func TestUnmarshalLogs(t *testing.T) {
 	expectedMinimum := plog.NewLogs()
 	resourceLogs := expectedMinimum.ResourceLogs().AppendEmpty()
 	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
-	scopeLogs.Scope().SetName("otelcol/" + typeStr)
```
Comment (codeboten) [inline receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler_test.go:239]: The scope name is changed here, this should be called out with a changelog entry

Comment (atoulme) [reply receiver/azureeventhubreceiver/azureresourcelogs_unmarshaler_test.go:239]: ok then, let's slow down and talk about it as part of #21382

**Review by djaglowski — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**